test/micro/org/openjdk/bench/java/math/BigIntegerPow.java

```diff
@@ -0,0 +1,157 @@
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.profile.GCProfiler;
+
+import java.math.BigInteger;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Thread)
+@Warmup(iterations = 1, time = 1)
+@Measurement(iterations = 1, time = 1)
+@Fork(value = 3)
+public class BigIntegerPow {
+
+    private BigInteger[] xsArray, sArray, mArray, lArray, xlArray;
+    private int xsExp, sExp, mExp, lExp, xlExp;
+    private static final int TESTSIZE = 1;
+
+    /*
+     * You can run this test via the command line:
+     *    $ mvn clean install
+     *    $ java -jar target/benchmarks.jar BigIntegerPow -prof gc
+     */
+    public static void main(String[] args) throws RunnerException {
+        Options opt = new OptionsBuilder()
+                .include(BigIntegerPow.class.getSimpleName())
+                .addProfiler(GCProfiler.class)
+                .build();
+
+        new Runner(opt).run();
+    }
+
+    @Setup
+    public void setup() {
+        Random r = new Random(1123);
+
+        xsExp = (1 << 20) - 1;
+        xsArray = new BigInteger[TESTSIZE]; /*
+         * Each array entry is atmost 64 bits
+         * in size
+         */
+        sExp = (1 << 18) - 1;
+        sArray = new BigInteger[TESTSIZE]; /*
+         * Each array entry is atmost 256 bits
+         * in size
+         */
+        mExp = (1 << 16) - 1;
+        mArray = new BigInteger[TESTSIZE]; /*
+         * Each array entry is atmost 1024 bits
+         * in size
+         */
+        lExp = (1 << 14) - 1;
+        lArray = new BigInteger[TESTSIZE]; /*
+         * Each array entry is atmost 4096 bits
+         * in size
+         */
+        xlExp = (1 << 12) - 1;
+        xlArray = new BigInteger[TESTSIZE]; /*
+         * Each array entry is atmost 16384 bits
+         * in size
+         */
+
+        for (int i = 0; i < TESTSIZE; i++) {
+            xsArray[i] = new BigInteger(64, r);
+            sArray[i] = new BigInteger(256, r);
+            mArray[i] = new BigInteger(1024, r);
+            lArray[i] = new BigInteger(4096, r);
+            xlArray[i] = new BigInteger(16384, r);
+        }
+    }
+
+    /** Test BigInteger.pow() with numbers long at most 64 bits  */
+    @Benchmark
+    @OperationsPerInvocation(TESTSIZE)
+    public void testPowXS(Blackhole bh) {
+        for (BigInteger s : xsArray) {
+            bh.consume(s.pow(xsExp));
```

Comment:
```suggestion
            bh.consume(s.pow(xsExp & ~0b0111_0101));
            bh.consume(s.pow(xsExp));
            bh.consume(s.pow(xsExp + 1));
```
or something similar, just to measure other than "all-ones" exponents as well.